pinot-common/src/main/java/org/apache/pinot/common/function/FunctionInvoker.java

```diff
@@ -131,4 +133,19 @@ public Object invoke(Object[] arguments) {
           "Caught exception while invoking method: " + _method + " with arguments: " + Arrays.toString(arguments), e);
     }
   }
+
+  /**
+   * Whether method is a scalar function with nullableParameters property set to true.
+   */
+  public boolean hasNullableParameters() {
+    for (final Annotation annotation : _method.getAnnotations()) {
+      if (annotation.annotationType().equals(ScalarFunction.class)) {
+        final ScalarFunction scalarFunctionAnnotation = (ScalarFunction) annotation;
+        if (scalarFunctionAnnotation.nullableParameters()) {
+          return true;
+        }
+      }
+    }
+    return false;
```

Comment:
I think the following would be cleaner (and probably more efficient)
```java
return _method.isAnnotationPresent(ScalarFunction.class) && _method.getAnnotation(ScalarFunction.class).nullableParameters();
```

Reply:
The annotation parsing logic can be moved to `FunctionRegistry`, and the nullable info can be stored in `FunctionInfo`